[fontconfig] Build error on wasm32-emscripten

### Operating system

MacOS 14.4.1 (Intel) and Windows 10

### Compiler

`em++`

### Steps to reproduce the behavior

```Shell
`vcpkg install fontconfig:wasm32-emscripten`
```

### Failure logs

Computing installation plan...
The following packages will be built and installed:
    fontconfig:wasm32-emscripten@2.15.0#1
Detecting compiler hash for triplet wasm32-emscripten...
Compiler found: /Users/mattiamontanari/repos/emsdk/upstream/emscripten/em++
Restored 0 package(s) from /Users/mattiamontanari/.cache/vcpkg/archives in 13.4 us. Use --debug to see more details.
Installing 1/1 fontconfig:wasm32-emscripten@2.15.0#1...
Building fontconfig:wasm32-emscripten@2.15.0#1...
/Users/mattiamontanari/repos/vcpkg/triplets/community/wasm32-emscripten.cmake: info: loaded community triplet from here. Community triplets are not built in the curated registry and are thus less likely to succeed.
-- Found Python version '3.11.7 at /opt/anaconda3/bin/python3'
-- Using meson: /Users/mattiamontanari/repos/vcpkg/downloads/tools/meson-1.6.0-ac58a9/meson.py
-- Using cached fontconfig-fontconfig-2.15.0.tar.gz.
-- Cleaning sources at /Users/mattiamontanari/repos/vcpkg/buildtrees/fontconfig/src/2.15.0-0e63e07859.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/mattiamontanari/repos/vcpkg/downloads/fontconfig-fontconfig-2.15.0.tar.gz
-- Applying patch no-etc-symlinks.patch
-- Applying patch libgetopt.patch
-- Applying patch fix-wasm-shared-memory-atomics.patch
-- Using source at /Users/mattiamontanari/repos/vcpkg/buildtrees/fontconfig/src/2.15.0-0e63e07859.clean
-- Found external ninja('1.12.1').
-- Configuring wasm32-emscripten-dbg
-- Getting CMake variables for wasm32-emscripten
-- Configuring wasm32-emscripten-dbg done
-- Configuring wasm32-emscripten-rel
-- Configuring wasm32-emscripten-rel done
-- Package wasm32-emscripten-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: /usr/local/bin/ninja install -v
    Working Directory: /Users/mattiamontanari/repos/vcpkg/buildtrees/fontconfig/wasm32-emscripten-dbg
    Error code: 1
    See logs for more information:
      /Users/mattiamontanari/repos/vcpkg/buildtrees/fontconfig/package-wasm32-emscripten-dbg-out.log

Call Stack (most recent call first):
  installed/x64-osx/share/vcpkg-tool-meson/vcpkg_install_meson.cmake:33 (vcpkg_execute_required_process)
  ports/fontconfig/portfile.cmake:41 (vcpkg_install_meson)
  scripts/ports.cmake:196 (include)


error: building fontconfig:wasm32-emscripten failed with: BUILD_FAILED
See https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle fontconfig:wasm32-emscripten: 1.1 min
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+fontconfig
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[fontconfig]+Build+error+on+wasm32-emscripten&body=Copy+issue+body+from+%2FUsers%2Fmattiamontanari%2Frepos%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md




### Additional context

_No response_

--- a/ports/fontconfig/emscripten.diff
+++ b/ports/fontconfig/emscripten.diff
@@ -1,0 +1,13 @@
+diff --git a/meson.build b/meson.build
+index 08d9532..37cc195 100644
+--- a/meson.build
++++ b/meson.build
+@@ -289,7 +289,7 @@ if fc_cachedir in ['yes', 'no', 'default']
+   endif
+ endif
+ 
+-if host_machine.system() != 'windows'
++if host_machine.system() != 'windows' and host_machine.system() != 'emscripten'
+   thread_dep = dependency('threads')
+   conf.set('HAVE_PTHREAD', 1)
+   deps += [thread_dep]

--- a/ports/fontconfig/portfile.cmake
+++ b/ports/fontconfig/portfile.cmake
@@ -17,6 +17,11 @@ if("nls" IN_LIST FEATURES)
 else()
     list(APPEND options "-Dnls=disabled")
 endif()
+if("tools" IN_LIST FEATURES)
+    list(APPEND options "-Dtools=enabled")
+else()
+    list(APPEND options "-Dtools=disabled")
+endif()
 
 vcpkg_configure_meson(
     SOURCE_PATH "${SOURCE_PATH}"
@@ -91,10 +96,12 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
     endforeach()
 endif()
 
-vcpkg_copy_tools(
-    TOOL_NAMES fc-match fc-cat fc-list fc-pattern fc-query fc-scan fc-cache fc-validate fc-conflist
-    AUTO_CLEAN
-)
+if("tools" IN_LIST FEATURES)
+    vcpkg_copy_tools(
+        TOOL_NAMES fc-match fc-cat fc-list fc-pattern fc-query fc-scan fc-cache fc-validate fc-conflist
+        AUTO_CLEAN
+    )
+endif()
 
 configure_file("${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake.in" "${CURRENT_PACKAGES_DIR}/share/${PORT}/vcpkg-cmake-wrapper.cmake" @ONLY)
 file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")

--- a/ports/fontconfig/portfile.cmake
+++ b/ports/fontconfig/portfile.cmake
@@ -33,7 +33,7 @@ vcpkg_configure_meson(
         -Diconv=enabled
         -Dtests=disabled
     ADDITIONAL_BINARIES
-        "gperf = ['${CURRENT_HOST_INSTALLED_DIR}/tools/gperf/gperf${HOST_EXECUTABLE_SUFFIX}']"
+        "gperf = ['${CURRENT_HOST_INSTALLED_DIR}/tools/gperf/gperf${VCPKG_HOST_EXECUTABLE_SUFFIX}']"
 )
 
 # https://www.freedesktop.org/software/fontconfig/fontconfig-user.html

--- a/ports/fontconfig/portfile.cmake
+++ b/ports/fontconfig/portfile.cmake
@@ -11,15 +11,23 @@ vcpkg_from_gitlab(
         fix-wasm-shared-memory-atomics.patch
 )
 
-vcpkg_add_to_path(PREPEND "${CURRENT_HOST_INSTALLED_DIR}/tools/gperf")
+set(options "")
+if("nls" IN_LIST FEATURES)
+    list(APPEND options "-Dnls=enabled")
+else()
+    list(APPEND options "-Dnls=disabled")
+endif()
 
 vcpkg_configure_meson(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
+        ${options}
         -Ddoc=disabled
         -Dcache-build=disabled
         -Diconv=enabled
         -Dtests=disabled
+    ADDITIONAL_BINARIES
+        "gperf = ['${CURRENT_HOST_INSTALLED_DIR}/tools/gperf/gperf${HOST_EXECUTABLE_SUFFIX}']"
 )
 
 # https://www.freedesktop.org/software/fontconfig/fontconfig-user.html
@@ -42,7 +50,7 @@ vcpkg_install_meson(ADD_BIN_TO_PATH)
 
 vcpkg_copy_pdbs()
 #Fix missing libintl static dependency
-if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
+if("nls" IN_LIST FEATURES AND VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
     if(NOT VCPKG_BUILD_TYPE)
         vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/fontconfig.pc" "-liconv" "-liconv -lintl" IGNORE_UNCHANGED)
     endif()

--- a/ports/fontconfig/portfile.cmake
+++ b/ports/fontconfig/portfile.cmake
@@ -106,4 +106,4 @@ endif()
 
 configure_file("${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake.in" "${CURRENT_PACKAGES_DIR}/share/${PORT}/vcpkg-cmake-wrapper.cmake" @ONLY)
 file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
-file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/COPYING")

--- a/ports/fontconfig/portfile.cmake
+++ b/ports/fontconfig/portfile.cmake
@@ -6,6 +6,7 @@ vcpkg_from_gitlab(
     SHA512 daa6d1e6058e12c694d9e1512e09be957ff7f3fa375246b9d13eb0a8cf2f21e1512a5cabe93f270e96790e2c20420bf7422d213e43ab9749da3255286ea65a7c
     HEAD_REF master
     PATCHES
+        emscripten.diff
         no-etc-symlinks.patch
         libgetopt.patch
         fix-wasm-shared-memory-atomics.patch

--- a/ports/fontconfig/usage
+++ b/ports/fontconfig/usage
@@ -1,4 +1,9 @@
 fontconfig is compatible with built-in CMake targets:
 
-    find_package(Fontconfig REQUIRED) # since CMake 3.14
-    target_link_libraries(main PRIVATE Fontconfig::Fontconfig)
+  find_package(Fontconfig REQUIRED) # since CMake 3.14
+  target_link_libraries(main PRIVATE Fontconfig::Fontconfig)
+
+fontconfig provides pkg-config modules:
+
+  # Font configuration and customization library
+  fontconfig

--- a/ports/fontconfig/vcpkg.json
+++ b/ports/fontconfig/vcpkg.json
@@ -42,6 +42,10 @@
         },
         "gettext-libintl"
       ]
+    },
+    "tools": {
+      "description": "Build tools",
+      "supports": "!emscripten"
     }
   }
 }

--- a/ports/fontconfig/vcpkg.json
+++ b/ports/fontconfig/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "fontconfig",
   "version": "2.15.0",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Library for configuring and customizing font access.",
   "homepage": "https://www.freedesktop.org/wiki/Software/fontconfig",
   "license": "MIT",
@@ -10,7 +10,6 @@
     "expat",
     "freetype",
     "getopt",
-    "gettext",
     {
       "name": "gperf",
       "host": true
@@ -21,12 +20,28 @@
     },
     {
       "name": "libuuid",
-      "platform": "!windows & !osx & !mingw"
+      "platform": "!osx & !windows"
     },
     "pthread",
     {
       "name": "vcpkg-tool-meson",
       "host": true
     }
-  ]
+  ],
+  "features": {
+    "nls": {
+      "description": "Native languages support",
+      "dependencies": [
+        {
+          "name": "gettext",
+          "host": true,
+          "default-features": false,
+          "features": [
+            "tools"
+          ]
+        },
+        "gettext-libintl"
+      ]
+    }
+  }
 }

--- a/ports/fontconfig/vcpkg.json
+++ b/ports/fontconfig/vcpkg.json
@@ -22,7 +22,10 @@
       "name": "libuuid",
       "platform": "!osx & !windows"
     },
-    "pthread",
+    {
+      "name": "pthread",
+      "platform": "!emscripten & !windows"
+    },
     {
       "name": "vcpkg-tool-meson",
       "host": true

--- a/ports/fontconfig/vcpkg.json
+++ b/ports/fontconfig/vcpkg.json
@@ -5,10 +5,14 @@
   "description": "Library for configuring and customizing font access.",
   "homepage": "https://www.freedesktop.org/wiki/Software/fontconfig",
   "license": "MIT",
+  "supports": "!uwp",
   "dependencies": [
     "dirent",
     "expat",
-    "freetype",
+    {
+      "name": "freetype",
+      "default-features": false
+    },
     {
       "name": "gperf",
       "host": true

--- a/ports/fontconfig/vcpkg.json
+++ b/ports/fontconfig/vcpkg.json
@@ -9,7 +9,6 @@
     "dirent",
     "expat",
     "freetype",
-    "getopt",
     {
       "name": "gperf",
       "host": true
@@ -48,7 +47,10 @@
     },
     "tools": {
       "description": "Build tools",
-      "supports": "!emscripten"
+      "supports": "!emscripten",
+      "dependencies": [
+        "getopt"
+      ]
     }
   }
 }

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2890,7 +2890,7 @@
     },
     "fontconfig": {
       "baseline": "2.15.0",
-      "port-version": 1
+      "port-version": 2
     },
     "foonathan-lexy": {
       "baseline": "2022.12.1",

--- a/versions/f-/fontconfig.json
+++ b/versions/f-/fontconfig.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "82b83bc494dd4915a6b830092705100268a22734",
+      "version": "2.15.0",
+      "port-version": 2
+    },
+    {
       "git-tree": "443f2cc8005cd715e8786521c1c2cd990a1320a8",
       "version": "2.15.0",
       "port-version": 1

--- a/versions/f-/fontconfig.json
+++ b/versions/f-/fontconfig.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "9dc0f11f020439862ef3e2aa0eb75177562f6f07",
+      "git-tree": "4e1bd69ece93d01f14f3f78da93cfeca0f7e8b5f",
       "version": "2.15.0",
       "port-version": 2
     },

--- a/versions/f-/fontconfig.json
+++ b/versions/f-/fontconfig.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "251a3ee32f314891381a4d8199d24cc5b8b45900",
+      "git-tree": "9dc0f11f020439862ef3e2aa0eb75177562f6f07",
       "version": "2.15.0",
       "port-version": 2
     },

--- a/versions/f-/fontconfig.json
+++ b/versions/f-/fontconfig.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "82b83bc494dd4915a6b830092705100268a22734",
+      "git-tree": "251a3ee32f314891381a4d8199d24cc5b8b45900",
       "version": "2.15.0",
       "port-version": 2
     },

